torchtitan/experiments/autoparallel/deepseek_v3/parallelize_deepseekv3.py

```diff
@@ -326,7 +335,7 @@ def input_fn():
     with AutoParallel(
         model,
         input_fn,
-        world_mesh,
+        dense_mesh,
```

Comment:
We need to pass the `sparse_mesh` here. For the dense part of the model we will do FSDP on `efsdp` and `ep` meshes and for the sparse part of the model we will do FSDP on `efsdp` mesh and Ep on the `ep` mesh. Additionally in AutoParallel we require TP=ETP, so we need to add a validation check here.